### ISSUE TITLE
Internal extension repository source support

### ISSUE THREAD

**Issue by BitFlippedCelt** [tilt-dev/tilt#4707]
## Describe the Feature You Want
Allow for the ability to source extensions for third-party/internal sources.

### Current Behavior
Tiltfile load() appears to only support loading from the official GitHub extensions registry.

### Why Do You Want This?
As my team and I have worked toward moving some of our projects to Tilt's local development platform, we have run into complexities that have accumulated over time that we need to initially port over into our evolving Tilt workflow so that we can later schedule some tech debt work against removing these patterns. Until that time though we need a way to manage these internal requirements in a way that would integrate with the existing Tilt extension loading pattern.

I have not started to investigate this from the Tilt codebase but I expect to start poking at it as I am able :)

**Comment by nicks** [comment]
as a workaround, some teams we work with use the `git_resource` extension to load a git repository somewhere on disk.

https://github.com/tilt-dev/tilt-extensions/tree/master/git_resource

Then, you can use the normal load() or load_dynamic() built-in to load extensions from that repo

But medium-term, we do want to have an easier way to load extensions from an arbitrary repository, stay tuned!

**Comment by BitFlippedCelt** [comment]
@nicks Thank you for the follow-up here!

I've been digging through the code base on my fork trying to see what I can decipher along this thread. I've been able to implement a HTTP(s) fetch mechanism (no auth, or anything fancy) that seems to be working, also nicely lays out fetched assets following the go-get pattern. When I started to poke at alternative URI types such as file:// and git:// though I started to run into issues with the go-get library and how it's pre-filtering the URLs. Was looking to maybe continue poking here but may just peel off to leverage `git_resource` to get over the hump and see if this allows us to move forward with migrating functionality.
